compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs

```diff
@@ -2090,10 +2090,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             && let Some(callee_ty) = callee_ty
         {
             let callee_ty = callee_ty.peel_refs();
+            let hir_id = self.tcx.hir().local_def_id_to_hir_id(self.body_id);
             match *callee_ty.kind() {
                 ty::Param(param) => {
                     let param =
-                        self.tcx.generics_of(self.body_id.owner).type_param(&param, self.tcx);
+                        self.tcx.generics_of(hir_id.owner).type_param(&param, self.tcx);
```

Comment:
`generics_of(hir_id.owner)` isn't right as the owner of a `HirId` is not the innermost context of the `hir_id`. Directly use `self.tcx.generics_of(self.body_id)` here